tm2/pkg/p2p/base_reactor.go

```diff
@@ -16,7 +16,7 @@ type Reactor interface {
 	service.Service // Start, Stop
 
 	// SetSwitch allows setting a switch.
-	SetSwitch(*Switch)
+	SetSwitch(Switch)
```

Comment:
That method does not define the needed functionality for Reactors, it doesn't make sense to have it here... Depending on the Reactor implementation you will use a Switch or not.